esx_service/utils/auth_data_test.py

```diff
@@ -0,0 +1,324 @@
+    #                                           default_privileges, vms, privileges)
+    #     self.assertTrue(uuid.UUID(tenant2.id))
+        
+    #     privileges = [{'datastore': 'datastore1',
+    #                    'global_visibility': 0,
+    #                    'create_volume': 0,
+    #                    'delete_volume': 0,
+    #                    'mount_volume': 0,
+    #                    'max_volume_size': 0,
+    #                    'usage_quota': 0}]
+
+    #     tenant1.set_datastore_access_privileges(self.auth_mgr.conn, privileges)
+
+    #     privileges = [{'datastore': 'datastore1',
+    #                     'global_visibility': 0,
+    #                     'create_volume': 1,
+    #                     'delete_volume': 1,
+    #                     'mount_volume': 0,
+    #                     'max_volume_size': 0,
+    #                     'usage_quota': 0}]
+        
+    #     tenant1.set_datastore_access_privileges(self.auth_mgr.conn, privileges)
+    
+
+    # def test_list_tenants(self):
+    #     vms = [(str(uuid.uuid4()), 'vm1')]
+    #     privileges = []
+    #     default_datastore = 'default_ds'
+    #     default_privileges = [{'datastore': default_datastore,
+    #                           'global_visibility': 0,
+    #                           'create_volume': 0,
+    #                           'delete_volume': 0,
+    #                           'mount_volume': 0,
+    #                           'max_volume_size': 0,
+    #                           'usage_quota': 0}]
+    #     tenant1 = self.auth_mgr.create_tenant('tenant1', 'Some tenant', default_datastore,
+    #                                           default_privileges, vms, privileges)
+    #     self.assertTrue(uuid.UUID(tenant1.id))
+        
+    #     vms = [(str(uuid.uuid4()), 'vm2'), (str(uuid.uuid4()), 'vm3')]
+    #     privileges = []
+    #     tenant2 = self.auth_mgr.create_tenant('tenant2', 'Some tenant', default_datastore,
+    #                                           default_privileges, vms, privileges)
+    #     self.assertTrue(uuid.UUID(tenant2.id))
+        
+    #     privileges = [{'datastore': 'datastore1',
+    #                    'global_visibility': 0,
+    #                    'create_volume': 0,
+    #                    'delete_volume': 0,
+    #                    'mount_volume': 0,
+    #                    'max_volume_size': 0,
+    #                    'usage_quota': 0}]
+
+    #     self.auth_mgr.set_datastore_access_privileges(tenant1, privileges)
+    #     tenants_list = self.auth_mgr.list_tenants()
+    #     for tenant in tenants_list:
+    #         print "Tenant Info Start"
+    #         print tenant.name
+    #         print tenant.description
+    #         print tenant.default_datastore
+    #         print tenant.default_privileges
+    #         print tenant.vms
+    #         print tenant.privileges
+    #         print "Tenant Info End"
```

Comment:
yes. Will refactor the code in auth_data_test.py accoding to your later comments.